core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -97,6 +98,11 @@ private[spark] object SizeEstimator extends Logging {
       return System.getProperty("spark.test.useCompressedOops").toBoolean
     }
 
+    // java.vm.info provides compressed ref info for IBM JDKs
+    if (System.getProperty("java.vendor").contains("IBM")) {
```

Comment:
How do you feel about me creating a getJavaVendor() utility in org.apache.spark.util.Utils rather than in-lining the checks?

It would, however, be a duplicate of the version proposed for the launcher which is not great.